src/components/Chip/ChipWithSelect.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {
+  InputHTMLAttributes,
+  OptionHTMLAttributes,
+  useState,
+} from 'react'
+import styled from '@emotion/styled'
+import { ChevronDown } from '../../icons'
+import { Chip, ChipProps } from './Chip'
+
+const Content = styled.span`
+  select {
+    position: absolute;
+    opacity: 0;
+    inset: 0;
+    cursor: pointer;
+  }
+`
+
+interface OptionProps extends OptionHTMLAttributes<HTMLOptionElement> {}
+
+type RequiredChipProps = Omit<ChipProps, 'active' | 'endAdornment'>
+
+interface ChipWithSelectProps
+  extends RequiredChipProps,
+    InputHTMLAttributes<HTMLSelectElement> {
+  options: OptionProps[]
+}
+
+export const ChipWithSelect = ({
+  startAdornment,
+  options,
+  value: externalValue,
+  onChange,
+  compact,
+  ...props
+}: ChipWithSelectProps) => {
+  const [internalValue, setInternalValue] = useState(
+    props.defaultValue || options[0].value
+  )
+  const value =
+    typeof externalValue !== 'undefined' ? externalValue : internalValue
+  const currentOption = options.find(
+    option => option.value === value
+  ) as OptionProps
+
+  return (
+    <Chip
+      compact={compact}
+      active={Boolean(value)}
+      startAdornment={startAdornment}
+      endAdornment={<ChevronDown size={16} />}
+      as="label"
+    >
+      <Content>
+        <select
+          value={value}
+          onChange={event => {
+            setInternalValue(event.target.value)
+            onChange?.(event)
+          }}
+          {...props}
+        >
+          {options.map((option, index) => (
+            <option key={index} {...option}>
```

Comment:
In the PR for the new Combobox we set the key like [this](https://github.com/TicketSwap/solar/pull/1615/files#diff-6ded7d0ece10dbe7a02884e7500135dfa041c03b2ea5acd98b3e50bb36de48caR132-R136). Maybe we can do that here as well?